torch/csrc/jit/node_hashing.cpp

```diff
@@ -19,27 +19,90 @@ bool tensorEqual(const at::Tensor& lhs, const at::Tensor& rhs) {
   return lhs.options().type_equal(rhs.options()) && lhs.equal(rhs);
 }
 
-bool tensorListEqual(
+bool typeListEqual(
+    const std::vector<TypePtr>& lhs,
+    const std::vector<TypePtr>& rhs) {
+  if (lhs.size() != rhs.size())
+    return false;
+  for (size_t i = 0; i < lhs.size(); ++i) {
+    if (*lhs[i] != *rhs[i]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+template <typename attribute_type> // int64_t, bool, double
+bool attributesEqual(attribute_type a1, attribute_type a2) {
```

Comment:
maybe `static_assert` that `attribute_type` is one of the supported types so this doesn't get used accidentally where it shouldn't

Reply:
i have a assert failure lower in the file, and there it's commented of when this function is okay to call into